doc/source/architecture.rst

```diff
@@ -3,4 +3,48 @@ Libreant architecture
 
 Libreant is meant to be a distributed system.
 Actually, you can even think of nodes as standalone-systems.
+A node is not aware of other nodes. It is a single point of distribution with no 
+knowledge of other points. 
 
+The system that binds the nodes together is the aggregator; an aggregator acts only
+as a client with respect to the nodes. Therefore multiple aggregators can coexist.
+This also implies that the node administration does not involve the management of 
+the aggregation mechanism and of the aggregators themselves.
+Similarly, it is possible to run an aggregator without running a libreant node.
+As a consequence, a node cannot choose whether to be aggregated or not.
+
+The aggregation mechanism is based on Opensearch_, and relies on two mandatory fields:
+
+ - the Opensearch description_
+
+ - the Opensearch response_
+
+meaning that this entries are mandatory on a node in order to be aggregated.
+The result component heavily relies on the relevance_ extension of the response spec.
+
+We blindly trust this relevance field, so a malicious node could bias the overall 
+result, simply increasing the relevance fields of its entries.
+In this way, the management of the aggregators implies also the task of checking
+the fairness of the aggregated nodes.
+
+
+How to set up an aggregator
+----------------------------
+
+1. Install Libreant. Follow the instructions on :ref:`dev-installation`.
+
+2. Launch Libreant setting the ``WEBANT_AGHERANT_DESCRIPTIONS`` environment variable. 
+   Its value should be a list of URLs encoded in JSON. Each URL represents the Opensearch 
+   description. For Libreant it's located in ``/description.xml``, so a typical URL looks
+   like::
+ 
+       http://127.0.0.1:5000/description.xml
+
+   and a typical invocation looks like::
+
+       WEBANT_AGHERANT_DESCRIPTIONS='["http://127.0.0.1:5000/description.xml","http://other.node/description.xml"]' webant   
```

Comment:
On Sun, Mar 29, 2015 at 02:52:32AM -0700, ael wrote:
>```http://127.0.0.1:5000/description.xml``` should be ```SELF```

you're right, I've fixed this.